src/modules/auth/auth.service.ts

```diff
@@ -53,20 +76,127 @@ export class AuthService {
         message: 'account already exists',
       };
     }
+
     const hashedPassword = await bcrypt.hash(input.password, 10);
+
+    const verificationToken =
+      this.emailVerificationService.generateVerificationToken(input.email);
     const res = await this.userService.create({
       ...input,
       password: hashedPassword,
+      isEmailVerified: false,
+      verificationToken,
     });
+    const registeredUser = await this.userService.findByEmail(input.email);
     if (res) {
+      await this.emailVerificationService.sendVerificationEmail(registeredUser);
       return {
         code: SUCCESS,
-        message: 'register successfully',
+        message:
+          'register successfully. An email has been sent to your email address for verification',
       };
     }
+
     return {
       code: REGISTER_ERROR,
       message: 'registration failed',
     };
   }
+
+  async verifyEmail(email: string, token: string): Promise<Result> {
+    try {
+      const emailRegex = /^[^\s@]+@[^\s@]+\.[^\s@]+$/;
```

Comment:
regex needs to be imported and keep the same through all different services for email pattern